SasView/sasview
# Rog and beta q

## Description

Adjusted the Generic Scattering Calculator to automatically calculate and display the Radius of Gyration (only for non-element nuclear data). Furthermore, added Beta Q calculations into the Generic Scattering Calculation. A new box has been added to the QComboBox, giving 3 options now- "Fixed Orientation" (2D), "Debye Full Avg." (1D) and "Full Avg. w/ Beta(Q)" (1D). Selecting Full Average w/ Beta Q will perform the same actions as Debye Full Average, but will also plot a second graph of "Q vs Beta(Q)" next to the "Q vs form factor" graph in the main window.

periodictable import has been added as well.

## How Has This Been Tested?

- Import Nuclear Data (ex. PDB File), preferably smaller or else the form factor calculation will take a long time
- Radius of Gyration should be displayed correctly, rounded to .1 Angstrom (checked against other calculations)
- Select "Full Avg. w/ Beta(Q)" from the Dropdown list and hit compute
- After computation is complete, two graphs should show up in the main window- Q vs Form Factor and Q vs Beta Q. (Beta Q Calculation has been checked against other calculations)

NOTE: Have not tested against Magnetic Data or Element Data yet due to a lack of Magnetic or Element Data files. May need to test to ensure the program is not crashed by it. May have to disable Beta Q Calculations for Magnetic Data if that is the case.

## Review Checklist (please remove items if they don't apply):

- [ ] Functionality has been tested
- [ ] User documentation is available and complete (if required)
- [ ] Developers documentation is available and complete (if required)
